library/n2_protocol.hpp

```diff
@@ -37,12 +59,30 @@ struct message {
 	virtual ~message() = default;
 };
 
+class call {
+public:
+	// to call on client side
+	virtual void request(std::unique_ptr<message> msg) = 0;
+	virtual std::unique_ptr<message> get_reply() = 0;
+	virtual int get_reply_error() const = 0;
```

Comment:
Why `get_reply_error` is `const`? `get_reply` and `get_reply_error` have one semantic.

Reply:
`get_reply` has reason to be non-const, but `get_reply_error` hasn't (and no reason to restrict `get_reply_error` to once-called). Semantic is alike, but not the same.

But it's true that const-method seems like outsider among other methods: I think, nobody will keep `const call`, because it's almost non-usable.